src/main/java/org/idpass/lite/IDPassReaderComponent.java

```diff
@@ -0,0 +1,143 @@
+package org.idpass.lite;
+
+import com.google.protobuf.ByteString;
+import io.mosip.kernel.core.util.CryptoUtil;
+import org.api.proto.Dat;
+import org.api.proto.Ident;
+import org.idpass.lite.exceptions.IDPassException;
+import org.idpass.lite.proto.PostalAddress;
+import org.springframework.stereotype.Component;
+
+import javax.imageio.ImageIO;
+import javax.imageio.ImageReadParam;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import com.github.jaiimageio.jpeg2000.impl.J2KImageReader;
+
+/**
+ * Spring boot singleton component execution wrapper of
+ * an IDPassReader instance
+ */
+
+@Component
+public class IDPassReaderComponent
+{
+    public IDPassReader reader;
+
+    /**
+     * Instantiates IDPassReader reader with a particular configuration
+     *
+     * @throws IDPassException Standard exception
+     * @throws IOException Standard exception
+     */
+    public IDPassReaderComponent()
+            throws IDPassException, IOException
+    {
+        reader = new IDPassReader();
+        reader.setDetailsVisible(
+                IDPassReader.DETAIL_GIVENNAME |
+                IDPassReader.DETAIL_SURNAME |
+                IDPassReader.DETAIL_PLACEOFBIRTH);
+    }
+
+    /**
+     * Returns a PNG image QR code representation as a byte[] array,
+     * from the given inputs:
+     *
+     * @param cs The credential subject input json
+     * @param pincode The IDPASS LITE pin code
+     * @param photob64 A facial photo image
+     * @return Returns PNG QR code of the generated IDPASS LITE card
+     */
+    public byte[] generateQrCode(String cs, String pincode, String photob64)
+    {
+        // Parse credential subject input json to populate fields in IdentFields object
+        IdentFields idf = IdentFields.getInstance(cs);
+
+        Ident.Builder identBuilder = Ident.newBuilder()
+                .setPin(pincode);
+
+        String imageType = photob64.split(",")[0];
+        byte[] photo = CryptoUtil.decodeBase64(photob64.split(",")[1]);
```

Comment:
What is the first part? is it the image format ? if so, we should check it.

Reply:
Yes. The last time I checked, I it was "png" despite the image is "jp2". Will raise this to MOSIP.

Reply:
yes, create a ticket on MOSIP, and then put a comment to explain it with the link to the ticket.